bftengine/tests/testViewChange/testViewChange.cpp

```diff
@@ -343,6 +343,138 @@ TEST(testViewchangeSafetyLogic_test, computeRestrictions_two_prepare_certs_for_s
+  auto* ppMsg2 = new PrePrepareMsg(pRepInfo->primaryOfView(view2),
+                                   view2,
+                                   assignedSeqNumIgnored,
+                                   bftEngine::impl::CommitPath::SLOW,
+                                   false,
+                                   clientRequest2->size());
+  ppMsg2->addRequest(clientRequest2->body(), clientRequest2->size());
+  ppMsg2->finishAddingRequests();
+
+  // Here we generate a valid Prepare Certificate for clientRequest2 with
+  // View=1, but with Sequence number below the last Stable Sequence.
+  // This Prepare Certificate's associated PrePrepare must not be present
+  // in the restrictions array generated by ViewChangeSafetyLogic::computeRestrictions
+  PrepareFullMsg* pfMsg2 =
+      PrepareFullMsg::create(view2, assignedSeqNumIgnored, pRepInfo->primaryOfView(view2), buff, sizeof(buff));
+
+  ViewChangeMsg** viewChangeMsgs = new ViewChangeMsg*[N];
+  auto futureView = view2 + 1;
+
+  viewChangeMsgs[0] = new ViewChangeMsg(0, futureView, lastStableSeqNum);
+  viewChangeMsgs[1] = nullptr;
+  viewChangeMsgs[2] = new ViewChangeMsg(2, futureView, lastStableSeqNum);
+  viewChangeMsgs[3] = new ViewChangeMsg(3, futureView, 0);
+
+  // Add the first Prepare Certificate
+  // to the View Change Msg from Replica 1
+  viewChangeMsgs[0]->addElement(*pRepInfo,
+                                assignedSeqNum,
+                                ppMsg1->digestOfRequests(),
+                                ppMsg1->viewNumber(),
+                                true,
+                                ppMsg1->viewNumber(),
+                                pfMsg1->signatureLen(),
+                                pfMsg1->signatureBody());
+
+  // Add the second Prepare Certificate
+  // to the View Change Msg from Replica 3
+  viewChangeMsgs[3]->addElement(*pRepInfo,
+                                assignedSeqNumIgnored,
+                                ppMsg2->digestOfRequests(),
+                                ppMsg2->viewNumber(),
+                                true,
+                                ppMsg2->viewNumber(),
+                                pfMsg2->signatureLen(),
+                                pfMsg2->signatureBody());
+
+  auto VCS = ViewChangeSafetyLogic(N, F, C, &dummyVerifier_, PrePrepareMsg::digestOfNullPrePrepareMsg());
+
+  SeqNum min{}, max{};
+  VCS.computeRestrictions(viewChangeMsgs, VCS.calcLBStableForView(viewChangeMsgs), min, max, restrictions);
+
+  for (int i = 0; i < kWorkWindowSize; i++) {
+    if (i == assignedSeqNum - min) {
+      // Check that the sequence number above the last stable is considered
+      Assert(!restrictions[assignedSeqNum - min].isNull);
+      Assert(ppMsg1->digestOfRequests().toString() == restrictions[assignedSeqNum - min].digest.toString());
+    } else {
+      // Check that all others are NULL, and thus the Prepare Certificate
+      // for the sequence number below the last stable is ignored.
```

Comment:
can you add an explicit check for this, outside of the "else" statement?

Reply:
Yes, we could also check that after calculating 
`VCS.computeRestrictions(viewChangeMsgs, VCS.calcLBStableForView(viewChangeMsgs), min, max, restrictions);`
the value of "min" is above "assignedSeqNumIgnored".